src/pydvl/utils/parallel/backend.py

```diff
@@ -15,10 +17,89 @@
 
 T = TypeVar("T")
 
-_PARALLEL_BACKEND: Optional["RayParallelBackend"] = None
+_PARALLEL_BACKENDS: Dict[str, "BaseParallelBackend"] = {}
 
 
-class RayParallelBackend:
+class BaseParallelBackend(ABC):
+    """Abstract base class for all parallel backends"""
+
+    config: Dict[str, Any] = {}
+
+    @abstractmethod
+    def get(self, v: Any, *args, **kwargs):
+        ...
+
+    @abstractmethod
+    def put(self, v: Any, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def wrap(self, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def wait(self, v: Any, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def effective_n_jobs(self, n_jobs: Optional[int]) -> int:
+        ...
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {self.config}>"
+
+
+class SequentialParallelBackend(BaseParallelBackend):
+    """Class used to run jobs sequentiall and locally. It shouldn't
+    be initialized directly. You should instead call `init_parallel_backend`.
+
+    :param config: instance of :class:`~pydvl.utils.config.ParallelConfig` with number of cpus
+
+    :Example:
+
+    >>> from pydvl.utils.parallel.backend import SequentialParallelBackend
+    >>> from pydvl.utils.config import ParallelConfig
+    >>> config = ParallelConfig(backend="sequential")
+    >>> parallel_backend = SequentialParallelBackend(config)
```

Comment:
This feels a bit awkward: first I select the backend in a generic config class, then I need to choose the right class manually and pass it the config object with the backend. Since backends are not supposed to be instantiated manually, maybe it is not a big deal, but then the example should illustrate `init_parallel_backend()` instead of direct construction (which wouldn't work if we disable init with a metaclass as suggested above)